packages/transport/src/sessions/background.ts

```diff
@@ -219,8 +241,9 @@ export class SessionsBackground extends TypedEmitter<{
         return Promise.resolve(this.success({ descriptors: Object.values(this.descriptors) }));
     }
 
+    // todo: this is wrong, there may be 2 same sessions for different paths
     private getPathBySession({ session }: GetPathBySessionRequest) {
-        let path: string | undefined;
+        let path: PathInternal | undefined;
```

Comment:
Better readable? But this selects the first descriptor with the given session, not the last one as before. :man_shrugging: 
```suggestion
        const path = Object.keys(this.descriptors).find(
            pathKey => this.descriptors[pathKey]?.session === session,
        );
```